Jenkinsfile

```diff
@@ -37,34 +119,24 @@ pipeline {
-                            sh "rm -rf ${sshPubKeyFile}"
-                            sh "rm -rf ${sshPrivKeyFile}"
-                            sh "mv ${instancesYaml} provision/template.yaml"
-                            sh "chmod 600 ${params.Login}-key.pub"
-                            sh "chmod 600 ${params.Login}-key.priv"
-                            sh "chmod 777 provision/prepare_template"
-                            // set +x and set -x are workaround to not print user password in jenkins output log
-                            ansiColor('xterm') {
-                                sh "set +x && cd provision && terraform init -from-module ${params.orchestrator} && ./createcontrail \"--create\" \"${params.Login}\" \"${params.Password}\" \"${params.ProjectID}\" \"${params.domainName}\" \"${params.projectName}\" \"${params.networkName}\" \"../${params.Login}-key.pub\" \"../${params.Login}-key.priv\" \"${params.routerIP}\" \"${params.orchestrator}\" \"${params.branch}\" \"${params.flavor}\" \"${params.contrail_type}\" \"${params.patchset_ref}\" && set -x"
-                            }
+                        ensureMainDirectoryExists()
+                        ensureUserDirectoryExists()
+
+                        dirExistingResult = checkIfDirectoryExists("../${mainDirectoryName}/${params.Login}/${params.MachineName}")
+                        if(dirExistingResult == '1'){
+                            error("It seems that there are actually resources with that name.\nPlease destroy them first or use if you just forgot about them. :)")
                         } else {
+                            prepareMachineDirectory()
                             ansiColor('xterm') {
-                                sh "set +x && cd provision && terraform init -from-module ${params.orchestrator} && ./createcontrail \"--create\" \"${params.Login}\" \"${params.Password}\" \"${params.ProjectID}\" \"${params.domainName}\" \"${params.projectName}\" \"${params.networkName}\" \"./id_rsa.pub\" \"./id_rsa\" \"${params.routerIP}\" \"${params.orchestrator}\" \"${params.branch}\" \"${params.flavor}\" \"${params.contrail_type}\" \"${params.patchset_ref}\" && set -x"
+                                sh "set +x && cd provision && terraform init ../../${mainDirectoryName}/${params.Login}/${params.MachineName} && ./createcontrail \"--create\" \"${params.Login}\" \"${params.Password}\" \"${params.MachineName}\" \"${params.ProjectID}\" \"${params.domainName}\" \"${params.projectName}\" \"${params.networkName}\" \"../../${mainDirectoryName}/${params.Login}/${params.MachineName}/${params.Login}-key.pub\" \"../../${mainDirectoryName}/${params.Login}/${params.MachineName}/${params.Login}-key.priv\" \"${params.routerIP}\" \"${params.orchestrator}\" \"${params.branch}\" \"${params.flavor}\" \"${terraformStateFileName}\" \"${mainDirectoryName}\" \"${params.contrail_type}\" \"${params.patchset_ref}\" && set -x"
```

Comment:
`./createcontrail` has way to many positional arguments, which are often times long paths. Maybe it's time to refactor? This is super scary and begs for bugs.

Reply:
Maybe use .tfvars?

Reply:
As above.